### ISSUE TITLE
Relative URL for openIdConnectUrl fails validation

### ISSUE THREAD

**Issue by jblazek** [Mermade/oas-kit#60]
### Checklist

- [x] Validation: I believe my source definition is valid OpenAPI 3.0.x but the validator complains

### Detailed Description

When using a relative URL in the `openIdConnectUrl` field, validation fails. When using an absolute URL, validation passes.

[Sample gist](https://gist.githubusercontent.com/jblazek/5b7cad3e3ec8002c3bf1a6e80fcc8408/raw/deaca3e01fd401d34dc00a1ababa4ee773463c61/openid-relative-url.yaml)

According to the spec, [relative URLs](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#relative-references-in-urls) are allowed.  I did not try other URLs, but it's possible this issue exists for other properties.

I opened this similar issue in swagger-editor, but it seems like they are using this engine for validation, so I think the underlying issue may be with the validator in this project.

https://github.com/swagger-api/swagger-editor/issues/1709

I can close this related issue if that project does depend on this one.

#### Other stuff

I found out about this project through the openapi.tools website. Using Validation engine v2.11.5 (hosted)

**Comment by MikeRalphson** [comment]
Thanks for raising the issue and including an example.

The problem was actually in the OAS draft schema (which the validator uses as a fallback).

The `openIdConnectUrl` was defined as:

```yaml
                 },
                 "openIdConnectUrl": {
                    "type": "string",
                    "format": "url"
                 },
```

When it should have been `format: uriref`.

This explains the similar behaviour of swagger-editor / swagger-ui - they are presumably using the same schema to drive validation, though I believe they are **not** using this project's (hand written) validator.

We are waiting on a dependency to come out of beta before cutting a new release, but if your requirement is urgent I can go back to the 2.11.x branch and make a patch. Please let me know.

**Comment by jblazek** [comment]
This is not urgent for me, I can possibly use an absolute URL for now, or just wait until it is fixed to switch from my current security scheme to openIdConnect.

You mentioned the 'OAS draft schema', where is that located?

**Comment by MikeRalphson** [comment]
https://github.com/OAI/OpenAPI-Specification/pull/1270

**Comment by jblazek** [comment]
I looked at that PR, and it looks like this comment also called out the format mismatch, but must have been missed.

https://github.com/OAI/OpenAPI-Specification/pull/1270#issuecomment-320981503

**Comment by MikeRalphson** [comment]
Yes, with the PR still being a work-in-progress, not all of the comments have been incorporated yet. In addition, the schema [here](https://raw.githubusercontent.com/Mermade/swagger2openapi/mono/packages/openapi-parser/schemas/openapi-3.0.json) has a bunch of descriptions ported over from https://github.com/OAI/OpenAPI-Specification/pull/1236

**Comment by MikeRalphson** [comment]
The fix to the schema has been included in `swagger2openapi` `v3.0.x`. Thanks for your patience.
